strategy/json/sql/postgres/single_stream_strategy.go

```diff
@@ -43,32 +45,29 @@ func (s *SingleStreamStrategy) CreateSchema(tableName string) []string {
     event_id UUID NOT NULL,
     event_name VARCHAR(100) NOT NULL,
     payload JSON NOT NULL,
-    metadata JSONB NOT NULL,
+    metadata JSONB NOT NULL, 
+    aggregate_type VARCHAR(50) NOT NULL,
+	aggregate_id UUID NOT NULL,
+	aggregate_version SMALLINT NOT NULL,
     created_at TIMESTAMP(6) NOT NULL,
     PRIMARY KEY (no),
-    CONSTRAINT aggregate_version_not_null CHECK ((metadata->>'_aggregate_version') IS NOT NULL),
-    CONSTRAINT aggregate_type_not_null CHECK ((metadata->>'_aggregate_type') IS NOT NULL),
-    CONSTRAINT aggregate_id_not_null CHECK ((metadata->>'_aggregate_id') IS NOT NULL),
     UNIQUE (event_id)
 );`,
 		tableName,
 	)
-	statements[1] = fmt.Sprintf(
-		`CREATE UNIQUE INDEX ON %s
-((metadata->>'_aggregate_type'), (metadata->>'_aggregate_id'), (metadata->>'_aggregate_version'));`,
-		tableName,
-	)
-	statements[2] = fmt.Sprintf(
-		`CREATE INDEX ON %s
-((metadata->>'_aggregate_type'), (metadata->>'_aggregate_id'), no);`,
-		tableName,
-	)
+	statements[1] = fmt.Sprintf(`CREATE UNIQUE INDEX ON %s (aggregate_type, aggregate_id, no);`, tableName)
```

Comment:
```suggestion
	statements[1] = fmt.Sprintf(`CREATE UNIQUE INDEX ON %s (aggregate_id, aggregate_type, no);`, tableName)
```

Let's define `aggregate_id` first in the index to increase selectivity factor.

Reply:
https://github.com/hellofresh/goengine/pull/50/commits/dd289f24a9ba230113e938de5e7aca8df60194e4